aries/aries_vcx_core/src/wallet/askar/askar_record_wallet.rs

```diff
@@ -0,0 +1,129 @@
+        base_wallet::{record::Record, search_filter::SearchFilter, RecordWallet},
+        entry_tags::EntryTags,
+    },
+};
+
+#[async_trait]
+impl RecordWallet for AskarWallet {
+    async fn add_record(&self, record: Record) -> VcxCoreResult<()> {
+        let mut session = self.backend.session(self.profile.clone()).await?;
+        let tags: Option<Vec<AskarEntryTag>> = Some(record.tags().clone().into());
+
+        Ok(session
+            .insert(
+                record.category(),
+                record.name(),
+                record.value().as_bytes(),
+                tags.as_deref(),
+                None,
+            )
+            .await?)
+    }
+
+    async fn get_record(&self, category: &str, name: &str) -> VcxCoreResult<Record> {
+        let mut session = self.backend.session(self.profile.clone()).await?;
+
+        Ok(session
+            .fetch(category, name, false)
+            .await?
+            .ok_or_else(|| {
+                AriesVcxCoreError::from_msg(
+                    AriesVcxCoreErrorKind::WalletRecordNotFound,
+                    "record not found",
+                )
+            })
+            .map(TryFrom::try_from)??)
+    }
+
+    async fn update_record_tags(
+        &self,
+        category: &str,
+        name: &str,
+        new_tags: EntryTags,
+    ) -> VcxCoreResult<()> {
+        let mut session = self.backend.session(self.profile.clone()).await?;
+
+        let found = session.fetch(category, name, true).await?;
+
+        let askar_tags: Vec<AskarEntryTag> = new_tags.into();
+
+        match found {
+            Some(record) => Ok(session
+                .replace(category, name, &record.value, Some(&askar_tags), None)
+                .await?),
+            None => Err(AriesVcxCoreError::from_msg(
+                AriesVcxCoreErrorKind::WalletRecordNotFound,
+                "wallet record not found",
+            )),
+        }
+    }
+
+    async fn update_record_value(
+        &self,
+        category: &str,
+        name: &str,
+        new_value: &str,
+    ) -> VcxCoreResult<()> {
+        let mut session = self.backend.session(self.profile.clone()).await?;
+
+        let found = session.fetch(category, name, true).await?;
+
+        match found {
+            Some(record) => Ok(session
+                .replace(
+                    category,
+                    name,
+                    new_value.as_bytes(),
+                    Some(&record.tags),
+                    None,
+                )
+                .await?),
+            None => Err(AriesVcxCoreError::from_msg(
+                AriesVcxCoreErrorKind::WalletRecordNotFound,
+                "wallet record not found",
+            )),
+        }
+    }
+
+    async fn delete_record(&self, category: &str, name: &str) -> VcxCoreResult<()> {
+        let mut session = self.backend.session(self.profile.clone()).await?;
+        Ok(session.remove(category, name).await?)
+    }
+
+    #[allow(unreachable_patterns)]
```

Comment:
Why is this necessary?

Reply:
* `-F askar_wallet,vdrtools_wallet` requires all match patterns to be covered
* `-F askar_wallet` complains about unreachable patterns since there is only one